### ISSUE TITLE
Interpreting the loss functions

### ISSUE THREAD

**Issue by goutamyg** [RobinWu218/SimDINO#40]
Hi,

I am not sure if I fully understand the "comp" and "expa" terms that define compression and expanison losses used during training. I could not find these terms in the paper as well. Can you briefly explain the intuition behind using these terms? 

Also, can you please help me understand the difference between these DINO losses: dino_mcr_comp_loss, dino_mcr_expa_loss, dino_mcr_global_comp_loss, dino_mcr_loss? Thank you!

-Goutam

**Comment by RobinZe** [comment]
Here is my personal understanding:

- `dino_mcr_comp_loss` refers to the cosine similarity between the feature matrices output by the teacher network and the student network after passing through the DINO head. Notably, the feature matrix of the student network includes outputs from both global and local crops.
- As a component of `dino_mcr_comp_loss`, `dino_mcr_global_comp_loss` only incorporates the global crop outputs from the student network.
- `dino_mcr_expa_loss` corresponds to the so-called "coding rate regularization" mentioned in the paper. Under the default configuration, it is calculated using the mean values of the features from the student network and the teacher network.
- Finally, `dino_mcr_loss` is the weighted sum of `dino_mcr_comp_loss` and `dino_mcr_expa_loss`.

I hope this understanding is accurate and proves helpful to you.

**Comment by RobinWu218** [comment]
Sorry for my late reply @goutamyg. @RobinZe is correct. Thanks!
